libraries/chain/include/eosio/chain/abi_serializer.hpp

```diff
@@ -54,6 +55,9 @@ struct abi_serializer {
    type_name get_action_type(name action)const;
    type_name get_table_type(name action)const;
 
+   optional<map<uint64_t, string>::const_iterator>  find_error_message( uint64_t error_code )const;
```

Comment:
Why an optional iterator? Doesn't an iterator act as an optional anyway?

Reply:
You would have to compare to the internal `error_messages` field, which I guess the caller can do since it is public, but I didn't want them to be concerned about the implementation details of the `abi_serializer` struct.

I didn't want to just return an `optional<string>` since that would create unnecessary copies, but maybe I should just replace `find_error_message` and `get_error_message` with one function that returns an `optional<string>` to address the safety issues at the cost of unnecessary copies.

Reply:
Your right that is ugly. I do like the idea of the `optional<string>` return.